examples/google-analytics/app/routes/contact.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { Form, useSubmit } from "remix";
+import type { ActionFunction } from "remix";
+
+import * as gtag from "~/utils/gtags";
+
+export const action: ActionFunction = () => {
+  return "";
+};
+
+export default function Contact() {
+  const submit = useSubmit();
+
+  const handleSubmit = (
+    e: React.SyntheticEvent<
+      | HTMLFormElement
+      | HTMLButtonElement
+      | HTMLInputElement
+      | FormData
+      | URLSearchParams
+      | {
+          [name: string]: string;
+        }
+      | null
+    >
+  ) => {
+    e.preventDefault();
+
+    const target = e.target as typeof e.target & {
+      message: { value: string };
+    };
+
+    console.log("target", target.message.value);
+
+    gtag.event({
+      action: "submit_form",
+      category: "Contact",
+      label: target.message.value
+    });
+
+    submit(e.currentTarget, { replace: true, method: "post" });
```

Comment:
The `preventDefault` is unnecessary here because we will call your `onSubmit` handler for you and you can just add whatever else you'd like here.

Care to double check this works as you expect?

```suggestion
    const target = e.target as typeof e.target & {
      message: { value: string };
    };

    console.log("target", target.message.value);

    gtag.event({
      action: "submit_form",
      category: "Contact",
      label: target.message.value
    });
```